lib/train/transports/cisco_ios_connection.rb

```diff
@@ -65,7 +66,9 @@ def run_command_via_connection(cmd, &_data_handler)
 
       logger.debug('[SSH] waiting for prompt')
       until @buf =~ @prompt
-        raise BadEnablePassword if @buf =~ /Bad secrets/
+        if @buf =~ /Bad [secrets|password]/ || @buf =~ /Access denied/
```

Comment:
This regexp looks wrong to me. Not sure if you want an alternation or to escape the square brackets but I'm guessing the former:

```ruby
        if @buf =~ /Bad (secrets|password)/ || @buf =~ /Access denied/
```

and then at that point, they might as well merge:

```ruby
        if @buf =~ /Bad (secrets|password)|Access denied/
```